common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -303,17 +340,37 @@ boolean transfer(Stack<?> dst) {
         private DefaultHandle<?>[] elements;
         private final int maxCapacity;
         private int size;
+        private final AtomicInteger availableSharedCapacity;
 
         private volatile WeakOrderQueue head;
         private WeakOrderQueue cursor, prev;
 
-        Stack(Recycler<T> parent, Thread thread, int maxCapacity) {
+        Stack(Recycler<T> parent, Thread thread, int maxCapacity, int maxSharedCapacityFactor) {
             this.parent = parent;
             this.thread = thread;
             this.maxCapacity = maxCapacity;
+            availableSharedCapacity = new AtomicInteger(Math.max(maxCapacity / maxSharedCapacityFactor, LINK_CAPACITY));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Unnecessary use of fully qualified name 'Math.max' due to existing static import 'java.lang.Math.max' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUnnecessaryFullyQualifiedName)